src/webui/www/private/scripts/contextmenu.js

```diff
@@ -538,5 +541,122 @@ window.qBittorrent.ContextMenu = (function() {
+                    this.showItem('rename');
+                    this.showItem('delete');
+                    this.showItem('newSubscription');
+                    this.hideItem('newFolder');
+                    this.hideItem('updateAll');
+                    this.showItem('copyFeedURL');
+                }
+            }
+            else {
+                this.showItem('update');
+                this.showItem('markRead');
+                this.hideItem('rename');
+                this.showItem('delete');
+                this.hideItem('newSubscription');
+                this.hideItem('newFolder');
+                this.hideItem('updateAll');
+                this.showItem('copyFeedURL');
+            }
+        }
+    });
+
+    const RssArticleContextMenu = new Class({
+        Extends: ContextMenu
+    });
+
+    const RssDownloaderRuleContextMenu = new Class({
+        Extends: ContextMenu,
+        adjustMenuPosition: function(e) {
+            this.updateMenuItems();
+
+            // draw the menu off-screen to know the menu dimensions
+            this.menu.setStyles({
+                left: '-999em',
+                top: '-999em'
+            });
+            // position the menu
+            let xPosMenu = e.page.x + this.options.offsets.x - $('rssdownloaderpage').offsetLeft;
+            let yPosMenu = e.page.y + this.options.offsets.y - $('rssdownloaderpage').offsetTop;
+            if (xPosMenu + this.menu.offsetWidth > document.documentElement.clientWidth)
+                xPosMenu -= this.menu.offsetWidth;
+            if (yPosMenu + this.menu.offsetHeight > document.documentElement.clientHeight)
+                yPosMenu = document.documentElement.clientHeight - this.menu.offsetHeight;
+            if (xPosMenu < 0)
+                xPosMenu = 0;
+            if (yPosMenu < 0)
+                yPosMenu = 0;
+            this.menu.setStyles({
+                left: xPosMenu,
+                top: yPosMenu,
+                position: 'absolute',
+                'z-index': '2000'
+            });
+        },
+        updateMenuItems: function() {
+            let selectedRows = window.qBittorrent.RssDownloader.rssDownloaderRulesTable.selectedRowsIds();
+            if (selectedRows.length == 0) {
+                this.showItem('addRule');
+                this.hideItem('deleteRule');
+                this.hideItem('renameRule');
+                this.hideItem('clearDownloadedEpisodes');
+            }
+            else if (selectedRows.length == 1) {
+                this.showItem('addRule');
+                this.showItem('deleteRule');
+                this.showItem('renameRule');
+                this.showItem('clearDownloadedEpisodes');
+            }
+            else {
+                this.showItem('addRule');
+                this.showItem('deleteRule');
+                this.hideItem('renameRule');
+                this.showItem('clearDownloadedEpisodes');
+            }
```

Comment:
You can move the `addRule` line out of the conditionals since it's the same for all three cases.